### PR TITLE
added macOS support

### PR DESCRIPTION
**Pull request by sjsone**
I tried to compile this project on macOS and got into some hurdles. 

This pull request fixes some problems and adds small portion to the Readme.

### CODE REVIEW

File: isp55e0.c [hunk 1]
```diff
@@ -843,7 +843,7 @@ static void load_file(struct device *dev, struct content *info)
 		off_t remaining = statbuf.st_size - total_read;
 		int ret = read(fd, info->buf + total_read, remaining);
 		if (dev->debug)
-			printf("info->buf %p total_read %ld statbuf.st_size %ld remaining %ld read %d\n ", info->buf, total_read, statbuf.st_size, remaining, ret);
+			printf("info->buf %p total_read %lld statbuf.st_size %lld remaining %lld read %d\n ", info->buf, total_read, statbuf.st_size, remaining, ret);
```
Comment (frank-zago) [inline isp55e0.c]: This breaks on linux. You need to cast all 3 variables to long long.

Comment (sjsone) [reply isp55e0.c]: i added the cast to `long long`

**Comment by frank-zago** [comment]
That new patch must be squashed into the 2nd commit, which added the build issue.
Otherwise it's good.

**Comment by sjsone** [comment]
The commits are now squashed

**Pull request merged**